### PR TITLE
feat(admin): Add an admin script to delete an account.

### PR DESCRIPTION
**Pull request by rfk**
Not working yet, pushing for visibility; @jbuck was this the sort of thing you had in mind for https://bugzilla.mozilla.org/show_bug.cgi?id=1477806 ?

### CODE REVIEW

File: scripts/delete-account.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  P.each(process.argv.slice(1), email => {
+    return db.accountRecord(email).then(account => {
+      console.log("Found account record:")
+      console.log("    uid:", account.uid)
+      console.log("    email:", account.email)
+      const rl = readline.createInterface({
+        input: process.stdin,
+        output: process.stdout,
+      })
+      return new Promise((resolve, reject) => {
```
Comment (philbooth) [inline scripts/delete-account.js]: I think in practice it works out okay but looks like you're mixing a native promise in with bluebird stuff here, is that what you wanted to do?

Comment (rfk) [reply scripts/delete-account.js]: not what I wanted; thanks!

File: scripts/delete-account.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+            }
+          }
+          accountDeleteRoute.handler(mockRequest).then(resolve, reject)
+        })
+      })
+    });
+  }).then(() => {
+    console.log("done!")
+    db.close()
+    // Process won't shut down on its own and I don't know why...
```
Comment (philbooth) [inline scripts/delete-account.js]: Does it make a difference if you return the promise from `db.close`?

Comment (rfk) [reply scripts/delete-account.js]: Thanks, tried it just now, but sadly doesn't seem to help

Comment (philbooth) [reply scripts/delete-account.js]: Oh wait, it looks weird I know, but what happens if you return the top-level promise too?

```
return DB.connect(
```

I seem to remember having this problem in one of the amplitude scripts and this fixed it. Under the hood I guess node wraps the script in a big IIFE, and so you have to explicitly return the promise even though it looks like your script is running globally or something.

**Comment by jbuck** [comment]
@rfk yeah, this is great!

**Comment by rfk** [comment]
Alrighty, this appears to be working locally!

I figured out the issue with shutdown thanks to a nifty tool called [wtfnode](https://www.npmjs.com/package/wtfnode), which identified open sockets to ` 127.0.0.1:6379`, aka redis.  So there's a second commit here to close the redis connection pool when closing down the DB instance.

One thing I haven't added is any tests for the new script; sadly I'm out of time to work on it further, and I wanted to get it up so we can use it for the pending GPDR request.  @mozilla/fxa-devs r?

File: lib/db.js [hunk 3]
```diff
@@ -76,14 +75,19 @@ module.exports = (
...
 
   DB.connect = function (options) {
     return P.resolve(new DB(options))
   }
 
   DB.prototype.close = function () {
-    return P.resolve(this.pool.close())
+    const promises = [this.pool.close()]
+    if (this.redis) {
+      promises.push(this.redis.close())
```
Comment (philbooth) [inline lib/db.js]: Ah, nice one! This should also help us update mocha fwiw. The latest version sits there hanging until all async listeners have resolved....

**Comment by shane-tomlinson** [comment]
ref https://github.com/mozilla/fxa-auth-server/issues/2531 to update mocha

**Pull request merged**